---
classoption: "runningheads"
title: "CytokineFinder: identifying disease associated cytokines"
shorttitle: ""
# If the paper title is too long for the 
# running head, you can set an abbreviated 
# paper title here.
# shorttitle: "Abbreviated paper title"
authors: 
  - name: "Jeffrey Tang"
    inst: "1,2"
    orcid: "0000-1111-2222-3333"
  - name: "Amrit Singh"
    inst: "1,2"
    orcid: "1111-2222-3333-4444"
# First names are abbreviated in the running head.
# If there are more than two authors, 'et al.' is used.
authorrunning: "F. Author et al." 
institutes: 
  - name: "Department of Anesthesiology, Pharmacology & Therapeutics, The University of British Columbia, Vanouver, Canada"
  - name: "Centre for Heart Lung Innvation, St. Paul's Hosital, Vanouver, Canada"
abstract: | 
  The abstract should briefly summarize the contents of the paper in 150--250 words.
keywords: 
  - First keyword
  - Second keyword
  - Another keyword
bibliography: lncs-test.bib
# Use csl when using default citation_package for Pandoc citeproc
csl: https://www.zotero.org/styles/springer-lecture-notes-in-computer-science
# citations with arabic numbers is prefered when using natbib
natbiboptions: numbers

output: 
  rticles::lncs_article:
    citation_package: default # or natbib
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE,
                      out.extra = "")  # This forces knitr to label all figures.
```

# Introduction
Importance of identifying driver cytokines

Cytokines are a broad class of low-weight molecules that act on cellular signalling cascades relevant to human immunology, from anti-pathogen responses to tissue-damaging inflammation (Lin and Leonard 2019). These cascades influence a wide variety of immune and inflammatory responses, cytokines have become well recognized for their roles in allergy, asthma, and clinical immunology which have primarily been driven by studies on cell biology, T cell polarization, and cytokine measurement experiments (ie. immunoassays) (Ghosh and Mersha 2022). As cytokines are an initiation point for these response mechanisms, leveraging it as a therapeutic target is desirable for several autoimmune diseases. It is thus important to identify driver cytokines involved in these regulatory activities (Deckers et al. 2023). Immunoassays such as enzyme linked immunosorbent assay (ELISA) and Luminex xMAP, can directly measure cytokine release, however due to the transitory nature of cytokines, measuring abundance prove to be challenging (Jiang et al. 2021; Deckers et al. 2023).

Researchers have considered compiling information relevant to cytokine signalling targets using high-throughput transcriptomic techniques and characterising patterns of the transcriptome to provide better understanding of relevant genes influenced cytokine-receptor binding, then validating findings with protein-protein interaction assays to bypass the limitations of low-throughput from immunoassays (Jiang et al. 2021).

Though cytokines have a short half-life, the signalling crosstalk mechanisms induced are often longer lasting and can act as a proxy to measure cytokine activity (Jiang et al. 2021). For example, a phenomena called receptor trafficking whereby in certain cases of surface receptor activation by a cytokine, promotes receptor gene expression to drive receptor syntheses further enhancing cellular responsiveness while in some instances, can lead to degradation of receptors (Martinez-Fabregas et al. 2019). Therefore, regulation of surface receptor expression is a dynamic process that involves a balance between receptor trafficking mechanisms and cytokine exposure. Thus surface receptor expression levels are an important factor to consider that can be used to identify cytokine activity levels (Farhat et al. 2021).

To infer on cytokine activity using receptor expression, we leverage publicly available high-throughput transcriptomic datasets and common strategies that utilise pre-defined gene sets that can be aggregated from catalogues of documented ligand-receptor interactions (ie. databases). We highlight 2 common strategies that rely on these annotations, gene set enrichment methods, gene summarization methods and a common discovery based method which considers building a model based on training data that are not reliant on annotations such as ligand-receptor interactions.


- mention limitations in publicly available cytokine data
- state hypothesis: using gene expression data we can identify the driver cytokines of disease

# Methods
## public cytokine data curation
  - cytokine stimulations
  - anti-cytokine treatment
  - studies with paired gene-expression and protein data
  
We selected studies that performed perturbation experiments to probe the response of cells given a conditional change. Perturbation-response profiling often employ molecular techniques to isolate a single cell into a controlled environment. As we are building on the assumption that intracellular effects of signalling cascades affect surface receptor gene expression, given the controlled environment and the response factor being a single cytokine treatment, we can assess how well receptor expression can capture cytokine activity using a variety of approaches in comparison to a de novo model-based method that is pre-trained on a large variety of data sets.

To test our hypothesis, we selected four different gene expression datasets, which were chosen for their variety of cytokines and easy accessibility. The first three datasets were all used in the same paper in parallel experiments, and can be found on GEO under GSE 135421, GSE 135425, and GSE 135426. These datasets compare healthy, unstimulated human neutrophils with those stimulated with one of three cytokines. They each contain six samples provided by three donors; the first sample per donor was created by stimulating the cells with IL-4, IFNg, and IL-13, respectively, while the last sample is a negative control.

The last dataset can be found on GEO under GSE 215039. This dataset compares healthy chondrocytes with osteoarthritis-affected chondrocytes when left untreated and when stimulated with one of three cytokines: IL-1B, IL-6, and TNF. This dataset contains 40 samples; 5 for each group mentioned above.

We also performed a literature search to gather different ligand-receptor databases. We selected a variety of databases which include the ligand-receptor interaction sets. We selected five arbitrarily due to the fact that there is an exhaustive list of databases that can be considered. Firstly, we selected a database generated by the Bader Lab group which developed an automated approach to construct a cell-cell interactions database and classified ligands and receptors as distinct proteins from gene ontology terms (Isserlin et al, 2020). We considered also considered CITEdb, a manually curated database for cell-cell interactions (Shan et al, 2022), FANTOM5 which is an atlas of enhancers, promoters, and other regulatory elements that capture gene regulatory networks of the transcriptome covering different cellular states (Kawaji et al, 2017), and NicheNet which is a method that predicts which ligands can influence expression in another cell using single cell expression data (Browaeys, Saelens, and Saeys 2020).


  
## study data
  - asthma gene expression data
  - validate with cytokien protein levels
  
## cytokneFinder
  - add workflow diagram of study (figure 1)
  - briefly describe each method; but add more details of each method in the supplement
  
## benchmarking analysis

## data analysis of study data
  - apply best method to study data

# Results

## Benchmarking (figure 2)
  - impact of annotations
  - which method+ann works best

## study results
   - patient demographics
   - cytokine finder results with study data
   - hypothesis testing for significant cytokines

# Discussion



## A Subsection Sample

Please note that the first paragraph of a section or subsection is not indented. 
The first paragraph that follows a table, figure, equation etc. does not need an indent, either.

Subsequent paragraphs, however, are indented.

### Sample Heading (Third Level)

Only two levels of headings should be numbered. 
Lower level headings remain unnumbered; they are formatted as run-in headings.

#### Sample Heading (Fourth Level)

The contribution should contain no more than four levels of headings. 
Table\ \ref{tab:table} gives a summary of all heading levels.

```{r table}
knitr::kable(head(pressure), 
             format = "latex",
             caption = "Table captions should be placed above the tables.")
```

\noindent Displayed equations are centered and set on a separate line.

\begin{equation}
x + y = z
\end{equation}

Please try to avoid rasterized images for line-art diagrams and schemas. 
Whenever possible, use vector graphics instead (see Fig.\ \ref{fig:fig1}).

```{r fig1, fig.cap = "A figure caption is always placed below the illustration. Please note that short captions are centered, while long ones are justified by the macro package automatically."}
plot(pressure)
```

::: {.theorem latex=true}
This is a sample theorem. The run-in heading is set in bold, while the following text appears in italics. Definitions, lemmas, propositions, and corollaries are styled the same way.
:::

The environments 'definition', 'lemma', 'proposition', 'corollary', 'remark', and 'example' are defined in the LLNCS documentclass as well.

::: {.proof latex=true}
Proofs, examples, and remarks have the initial word in italics, while the following text appears in normal font.
:::

For citations of references, use [@Nuncio2011].
Multiple citations are grouped [@Levitus2012], [@Raphael2004].

### Acknowledgements
Please place your acknowledgments at the end of the paper, preceded by an unnumbered run-in heading (i.e. 3rd-level heading).

# References {#references}