---
title: "Intro - how to use CytokineFindeR"
author: "Jeffrey Tang"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{cytokineFindeR}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

The following is a brief introduction on the usage of `CytokineFindeR`.

This vignette will outline the following procedures:
1) Setting up the design matrix for the linear models
2) Loading in the demo data `golimumab` from `GSE92415` on GEO. The data set is subset specifically for ulcerative colitis cell isolates looking at the gene expression difference for golimumab, which is an anti-TNF drug, between week 0 and week 6. 
3) Running the benchmark analysis in parallel using `future`
4) Data visualization which shows p-values 

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Load the R packages
```{r setup}
library(cytokineFindeR)
```

Load demo data (GSE92415) which holds a cleaned up expression set and meta data subset for the treatment (golimumab) on ulcerative colitus patients
```{r}
data(golimumab)

obs_id <- golimumab$obs_id
treatment <- golimumab$cond

eset <- golimumab$qc_eset
```

## Pre-filtered databases against CytoSig cytokines

The full archive of LRI reference databases is `data("dbs_all")`

For this demo, we will be using a subset containing 3 databases specific to cytosig cytokines.

```{r}
# load the default databases list of lists
data("dbs_cytosig")

# 
dbs <- dbs_cytosig[names(dbs_cytosig) %in% c("baderlab", "lianaplus", "cellchat")]
```

## Set up the design matrix for demo data 
```{r}
# Input treatment: difference between week 0 and week 6
# input obs_id if paired data (in this case we do have paired data), GSM IDs different for same sample IDs due to different time points

design <- create_design(treatment, obs_id, eset)
design_mat <- design$design
dupcor <- design$dupcor
```

## run preprocessing step
```{r}
preprocess_list <- preprocess_eset(eset, dbs)
eset_f <- preprocess_list$eset_f
dbs_f <- preprocess_list$dbs_f
```

## Call the databases, list out the methods to benchmark,  run `cytokinefinder()`

The result should return a nested list of benchmark results

```{r, warning=FALSE}
methods <- c("gsva_limma", 
             "pca_limma")

results <- run_lri_methods(eset_f,
                           design_mat,
                           dbs_f,
                           methods,
                           treatment = treatment,
                           obs_id = obs_id,
                           correlation = dupcor$consensus
                          )
```

## To reconstruct the list of benchmarks into a tibble:

`benchlist_to_tbl` will rank the cytokine of interest based on the name assignment provided in `golimumab_treatment`, in this case, as the benchmark results are named by `TNF`, the function will output the ranking of TNF across all method+db combinations.

```{r}
golimumab_treatment <- list(TNF = list(benchmarks = results))
golimumab_tibble <- benchlist_to_tbl(results_list = golimumab_treatment,
                                     study_type = "treatment")
golimumab_tibble
```


## CytoSig Custom ridge can be run through the following function:

The `cytosig_custom_ridge` function will run differential expression using limma and parse the logFC of all genes in the top table. 

The ridge regression implementation takes the common genes based on the beta coefficients from CytoSig and computes the cytokine signatures accordingly.

```{r}
data("cytosig_beta")
cytosig <- cytosig_custom_ridge(eset, 
                     design_mat, 
                     obs_id = obs_id,
                     correlation = dupcor$consensus, 
                     beta_coef = cytosig_beta)
cytosig
```

